blockchain/state_migration.go

```diff
@@ -52,20 +53,20 @@ func (td *stateTrieMigrationDB) ReadPreimage(hash common.Hash) []byte {
 	return td.ReadPreimageFromNew(hash)
 }
 
-func (bc *BlockChain) stateMigrationCommit(s *statedb.TrieSync, db database.DBManager) (int, time.Duration, error) {
-	start := time.Now()
-	stateTrieBatch := db.NewBatch(database.StateTrieDB)
-
-	written, err := s.Commit(stateTrieBatch)
+func (bc *BlockChain) stateMigrationCommit(s *statedb.TrieSync, batch database.Batch) (int, error) {
+	written, err := s.Commit(batch)
 	if written == 0 || err != nil {
-		return written, 0, err
+		return written, err
```

Comment:
Can we sure that `written` is always bigger than 0 in normal cases?

Reply:
Yes, it is.
`written` is `written := len(s.membatch.order)`.
So it is always 0 or more.

If the all result has children, 0 `written` is possible.